src/screens/SignupScreen.tsx

```diff
@@ -1,53 +1,117 @@
-import React from "react";
-import { StyleSheet, Dimensions, View, Text, TouchableOpacity } from "react-native";
-import { StackNavigationProp } from '@react-navigation/stack';
-import { LoggedOutParamList } from "../../types"
+import React, { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  StyleSheet,
+  Dimensions,
+  View,
+  Text,
+  TouchableOpacity,
+  TextInput,
+  Alert,
+} from "react-native";
+import { StackNavigationProp } from "@react-navigation/stack";
+import { LoggedOutParamList } from "../../types";
 import { colorPalette, shadowStyle } from "../constants/ColorPalette";
+import { firebaseApp, api } from "../constants/Firebase";
+import { setUser } from "../redux/actions";
 
 const _screen = Dimensions.get("screen");
 export interface SignupScreenProps {
-    navigation: StackNavigationProp<LoggedOutParamList, "SignupScreen">
+  navigation: StackNavigationProp<LoggedOutParamList, "SignupScreen">;
 }
 
 export default function SignupScreen({ navigation }: SignupScreenProps) {
+  const [email, setEmail] = useState<string | null>(null);
+  const [password, setPassword] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const userState = useSelector<RootState, UserState>(
+    (state) => state.userState
+  );
+  const dispatch = useDispatch();
 
-    return (
-        <View style={styles.container}>
-            <Text> Signup Screen </Text>
-            <View style={styles.subContainer}>
-                <TouchableOpacity
-                    onPress={() => {
+  useEffect(() => {
+    if (error !== null) {
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+  }, [error]);
 
-                        // TODO: 
-                        // - Register user by making API request
-                        navigation.goBack()
-                    }}
-                >
-                    <Text>Register</Text>
+  const handleSignUp = async () => {
+    // Check that both email and password field are populated.
+    // Also check that the email is somewhat valid
+    if (email && password && /^\S+@\S+\.\S+$/.test(email)) {
+      try {
+        const user = await firebaseApp
+          .auth()
+          .createUserWithEmailAndPassword(email, password);
+        if (user.additionalUserInfo?.isNewUser) {
+          //TODO:
+          //Send user info to DB
+        }
+        //Update global state
+        dispatch(
+          setUser({
+            ...userState.user,
+            id: user?.user?.uid,
+            username: user?.user?.email,
+            image_url: user?.user?.photoURL,
+            isLoggedIn: true,
```

Comment:
I believe this can simply be 
```js
setUser({
id: resp.user?.uid,
username: resp.user?.email,
image_url: resp.user?.photoURL})
```

setUser already sets ```isLoggedIn: true``` and, the setUser action takes in a `User` object not a `UserState` object. `UserState` is what contains isLoggedIn not the `User`